### PR TITLE
Feature/add zendesk tasks

### PR DESCRIPTION
**Pull request by AlessandroLollo**
## Summary
This PR adds the ability to export tickets from Zendesk using the [Zendesk Incremental Export API](https://developer.zendesk.com/api-reference/ticketing/ticket-management/incremental_exports/#incremental-ticket-export).
Thanks @grandimk for the inspiration!

## Importance
Users can now export tickets and related entities from Zendesk using a native Prefect Task 😊 

This PR:

- [x] adds new tests (if appropriate)
- [x] adds a change file in the `changes/` directory (if appropriate)
- [x] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

**Comment by AlessandroLollo** [comment]
Here you can find the test error log https://app.circleci.com/pipelines/github/PrefectHQ/prefect/10632/workflows/f7c55589-441a-43ec-9724-ae0b09a37c12/jobs/139040/parallel-runs/0/steps/0-107

Not sure if it is related to the new Zendesk Task though...

File: src/prefect/tasks/zendesk/__init__.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"""
+This module contains a collection of tasks to export data from Zendesk
```
Comment (tpdorsey) [inline src/prefect/tasks/zendesk/__init__.py:2]:
```suggestion
This module contains a collection of tasks to export data from Zendesk.
```

File: src/prefect/tasks/zendesk/zendesk_tasks.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+import requests
+import time
+from typing import List, Union
+
+
+class ZendeskTicketsIncrementalExportTask(Task):
+    """
+    This task can be used to perform an incremental export of tickets from Zendesk.
+    More info at
+    https://developer.zendesk.com/api-reference/ticketing/ticket-management/incremental_exports/#incremental-ticket-export
```
Comment (tpdorsey) [inline src/prefect/tasks/zendesk/zendesk_tasks.py:18]:
```suggestion
    https://developer.zendesk.com/api-reference/ticketing/ticket-management/incremental_exports/#incremental-ticket-export.
```

File: src/prefect/tasks/zendesk/zendesk_tasks.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+    Args:
+        - subdomain (str, optional): The Zendesk subdomain to use to export tickets.
+        - email_address (str, optional): The email address to use to authenticate on Zendesk
```
Comment (tpdorsey) [inline src/prefect/tasks/zendesk/zendesk_tasks.py:22]: Make sure each sentence in your arguments descriptions ends with a period.

File: src/prefect/tasks/zendesk/zendesk_tasks.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        - api_token (str, optional): The API token to use to athenticate on Zendesk
+            If passed, it will take precedence over `api_token_env_var`.
+        - api_token_env_var (str, optional): The name of the env var which contains the
+            API token to use to authenticate on Zendesk.
+        - start_time (int, datetime, optional): The start time to use to export tickets.
+            Can be passed as an epoch timestamp or a `datetime` object.
+        - cursor (str, optional): The cursor to use to export tickets.
+            If passed, it will take precedence over `start_time`.
+        - exclude_deleted: (bool, optional): Whether to exclude deleted tickets or not.
```
Comment (tpdorsey) [inline src/prefect/tasks/zendesk/zendesk_tasks.py:31]: What is the default?

File: src/prefect/tasks/zendesk/zendesk_tasks.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+        - include_entities: (str, list, optional): Optional list of entities to side load.
+          More info at
+          https://developer.zendesk.com/documentation/ticketing/using-the-zendesk-api/side_loading/
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
```
Comment (tpdorsey) [inline src/prefect/tasks/zendesk/zendesk_tasks.py:35]:
```suggestion
        - **kwargs (dict, optional): Additional keyword arguments to pass to the
```

File: src/prefect/tasks/zendesk/zendesk_tasks.py [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+        start_time: Union[int, datetime] = None,
+        cursor: str = None,
+        exclude_deleted: bool = None,
+        include_entities: List[str] = None,
+    ):
+        """
+        Task run method to perform an incremental export of tickets from Zendesk.
+        Args:
+            - subdomain (str, optional): The Zendesk subdomain to use to export tickets.
+            - email_address (str, optional): The email address to use to authenticate on Zendesk
```
Comment (tpdorsey) [inline src/prefect/tasks/zendesk/zendesk_tasks.py:88]: As above, make sure each sentence ends with a period or other appropriate punctuation.

File: src/prefect/tasks/zendesk/zendesk_tasks.py [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+            - api_token (str, optional): The API token to use to athenticate on Zendesk
+                If passed, it will take precedence over `api_token_env_var`.
+            - api_token_env_var (str, optional): The name of the env var which contains the
+                API token to use to authenticate on Zendesk.
+            - start_time (int, datetime, optional): The start time to use to export tickets.
+                Can be passed as an epoch timestamp or a `datetime` object.
+            - cursor (str, optional): The cursor to use to export tickets.
+                If passed, it will take precedence over `start_time`.
+            - exclude_deleted: (bool, optional): Whether to exclude deleted tickets or not.
```
Comment (tpdorsey) [inline src/prefect/tasks/zendesk/zendesk_tasks.py:97]: Default?

File: src/prefect/tasks/zendesk/zendesk_tasks.py [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+        Raises:
+            - `ValueError` if both `api_token` and `api_token_env_var` are missing.
+            - `ValueError` if `api_token` is missing and `api_token_env_var` cannot be found.
+            - `ValueError` if `subdomain` is missing.
+            - `ValueError` if `email_address` is missing.
+            - `ValueError` if both `start_time` and `cursor` are missing.
+            - `prefect.engine.signals.FAIL` if the Zendesk API call fails.
+
+        Returns:
+            - a `dict` containing the list of tickets and, optionally, the included
```
Comment (tpdorsey) [inline src/prefect/tasks/zendesk/zendesk_tasks.py:111]:
```suggestion
            - A `dict` containing the list of tickets and, optionally, the included
```

**Review by tpdorsey — CHANGES_REQUESTED** [review]

File: tests/tasks/zendesk/test_zendesk_tasks.py [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+        )
+
+        zendesk_task.run(
+            subdomain="test",
+            email_address="foo@bar.com",
+            api_token="abc",
+            start_time=123,
+        )
+
+        assert "API rate limit reached!" in caplog.text
```
Comment (desertaxle) [inline tests/tasks/zendesk/test_zendesk_tasks.py:167]: Tests that check what is logged by the task are coupled to the specific log message. In order to make the test more robust, you could use the context manager implementation provided by responses (https://github.com/getsentry/responses#responses-as-a-context-manager). When using the context manager, responses will fail if not all the configured responses are invoked. That should ensure that this code is working correctly without needed to inspect the logs.

**Review by desertaxle — CHANGES_REQUESTED** [review]
Implementation looks good! I recommend refactoring the tests so that inspecting the logs isn't necessary by using the context manager implementation offered by `responses`.

**Comment by AlessandroLollo** [comment]
Error is
```
FAILED tests/core/test_flow.py::TestFlowRunMethod::test_flow_dot_run_updates_the_scheduled_start_time_of_each_scheduled_run
```

Not sure it is related to the new Zendesk task though...

File: src/prefect/tasks/zendesk/zendesk_tasks.py (hunk 4 above)
Comment (AlessandroLollo) [reply src/prefect/tasks/zendesk/zendesk_tasks.py:31]: I described the default value at line 32

File: changes/pr5276.yaml [hunk 10]
```diff
@@ -0,0 +1,5 @@
+feature:
```
Comment (desertaxle) [inline changes/pr5276.yaml:1]: Looks like this got pulled over from one of your other PRs. Can you remove it from this branch?

**Review by desertaxle — CHANGES_REQUESTED** [review]
These changes looks great! Just one small clean up comment.

**Review by tpdorsey — APPROVED** [review]
Looks good from a docs perspective. Thanks!

**Review by desertaxle — APPROVED** [review]

**Pull request merged**